File: gc-kettle/kettle-spring-boot-starter/src/main/java/com/gc/kettle/starter/pool/KettleRepositoryProvider.java
package com.gc.kettle.starter.pool;

import org.pentaho.di.repository.kdr.KettleDatabaseRepository;

/**
 * @author shizhongming
 * 2020/7/16 8:18 下午
 */
public interface KettleRepositoryProvider {

    /**
     * 获取资源库
     * @return kettle资源库
     */
    KettleDatabaseRepository getRepository();

    /**
     * 归还资源库连接池
     * @param repository kettle资源库
     */
    void returnRepository(KettleDatabaseRepository repository);
}

File: smart-starter/kettle-spring-boot-starter/src/main/kotlin/com/smart/starter/kettle/pool/KettleRepositoryProvider.kt
package com.smart.starter.kettle.pool

import org.pentaho.di.repository.kdr.KettleDatabaseRepository

/**
 *
 * @author ming
 * 2019/7/11 下午3:21
 */
interface KettleRepositoryProvider {

    fun getRepository(): KettleDatabaseRepository

    fun returnRepository(repository: KettleDatabaseRepository)
}